wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Like/Handler.ts

```diff
@@ -0,0 +1,306 @@
+        badgeClassNames: "",
+        isSingleItem: false,
+        markListItemAsActive: false,
+        renderAsButton: true,
+        summaryPrepend: true,
+        summaryUseIcon: true,
+
+        // permissions
+        canDislike: false,
+        canLike: false,
+        canLikeOwnContent: false,
+        canViewSummary: false,
+
+        // selectors
+        badgeContainerSelector: ".messageHeader .messageStatus",
+        buttonAppendToSelector: ".messageFooter .messageFooterButtons",
+        buttonBeforeSelector: "",
+        containerSelector: "",
+        summarySelector: ".messageFooterGroup",
+      },
+      opts,
+    ) as LikeHandlerOptions;
+
+    this.initContainers();
+
+    DomChangeListener.add(`WoltLabSuite/Core/Ui/Like/Handler-${objectType}`, () => this.initContainers());
+
+    new UiReactionHandler(this._objectType, {
+      containerSelector: this._options.containerSelector,
+    });
+  }
+
+  /**
+   * Initializes all applicable containers.
+   */
+  initContainers(): void {
+    let triggerChange = false;
+
+    document.querySelectorAll(this._options.containerSelector).forEach((element: HTMLElement) => {
+      if (this._containers.has(element)) {
+        return;
+      }
+
+      const elementData = {
+        badge: null,
+        dislikeButton: null,
+        likeButton: null,
+        summary: null,
+
+        dislikes: ~~element.dataset.likeDislikes!,
+        liked: ~~element.dataset.likeLiked!,
+        likes: ~~element.dataset.likeLikes!,
+        objectId: ~~element.dataset.objectId!,
+        users: JSON.parse(element.dataset.likeUsers!),
+      };
+
+      this._containers.set(element, elementData);
+      this._buildWidget(element, elementData);
+
+      triggerChange = true;
+    });
+
+    if (triggerChange) {
+      DomChangeListener.trigger();
+    }
+  }
+
+  /**
+   * Creates the interface elements.
+   */
+  protected _buildWidget(element: HTMLElement, elementData: ElementData): void {
+    let badgeContainer: HTMLElement | null;
+    let isSummaryPosition = true;
+
+    if (this._options.isSingleItem) {
+      badgeContainer = document.querySelector(this._options.summarySelector);
+    } else {
+      badgeContainer = element.querySelector(this._options.summarySelector);
+    }
+
+    if (badgeContainer === null) {
+      if (this._options.isSingleItem) {
+        badgeContainer = document.querySelector(this._options.badgeContainerSelector);
+      } else {
+        badgeContainer = element.querySelector(this._options.badgeContainerSelector);
+      }
+
+      isSummaryPosition = false;
+    }
+
+    if (badgeContainer !== null) {
+      const summaryList = document.createElement("ul");
+      summaryList.classList.add("reactionSummaryList");
+      if (isSummaryPosition) {
+        summaryList.classList.add("likesSummary");
+      } else {
+        summaryList.classList.add("reactionSummaryListTiny");
+      }
+
+      const availableReactions = new Map(Object.entries(window.REACTION_TYPES));
```

Comment:
I'd move this into the module declaration, because it only depends on data in `window`.